sdk/ai/ai-inference-rest/src/tracingPolicy.ts

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import {
+  type TracingClient,
+  type TracingContext,
+  type TracingSpan,
+  createTracingClient,
+} from "@azure/core-tracing";
+import { logger } from "./logger.js";
+import { getErrorMessage } from "@azure/core-util";
+import { tryProcessResponse, onStartTracing, tryProcessError, getSpanName } from "./traceHelper.js";
+import { PipelinePolicy, PipelineRequest, PipelineResponse, SendRequest } from "@azure/core-rest-pipeline";
+import { SDK_VERSION } from "./constants.js";
+
+/**
+ * The programmatic identifier of the tracingPolicy.
+ */
+export const tracingPolicyName = "inferenceTracingPolicy";
+
+/**
+ * A simple policy to create OpenTelemetry Spans for each request made by the pipeline
+ * that has SpanOptions with a parent.
+ * Requests made without a parent Span will not be recorded.
+ */
+export function tracingPolicy(): PipelinePolicy {
+
+  const tracingClient = tryCreateTracingClient();
+
+  return {
+    name: tracingPolicyName,
+    async sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse> {
+
+
+      const url = new URL(request.url);
+      if (!tracingClient || !url.href.endsWith("/chat/completions"))
+        return next(request);
+
+      const { span, tracingContext } = tryCreateSpan(tracingClient, request) ?? {};
+
+
+      if (!span || !tracingContext) {
+        return next(request);
+      }
+
+      try {
+        if (request.tracingOptions) {
```

Comment:
Instead of checking if the request has tracingOptions, I would just set them on the request

```ts
request.tracingOptions ??= {}
request.tracingOptions.tracingContext = tracingContext
```

It's not guaranteed that the request will have tracing options (if the user has not passed them in)